Add a new option during builder setup: 'Use single field'

mkickax opened the pull request:
So that the builder has a single field: the object to be built itself, instead of all fields of the object to be built. This requires less work to maintain builder when adding extra fields afterwards.

mjedynak commented:
@mkickax Unfortunately this approach does not work correctly if the class has no setters but is using constructor.
Generally I think that it would be better to implement https://github.com/mjedynak/builder-generator-idea-plugin/issues/18 - it would solve issue of 'maintaining the builder'. I see you're already familiar with the codebase so would really appreciate help in implementing it.

mkickax commented:
@mjedynak I see your point. I've not pay attention to the case of class using constructor and not setters. However, in my company we do use builders with single field pattern and we have plenty of them. Thus such an option will be very appreciated. Could a special mention such as "only for class with setters" alongside the option name do the trick?
Else I can use my fork and distribute it to my co-workers.

About #18, I don't really need it, but I can have a look at it, just for the challenge (no commitment from my side).  I will comment the issue in order to get more info.

mkickax commented:
@mjedynak what do you about changing `useSingleField = context.useSingleField();` by `useSingleField = context.useSingleField() && allSelectedPsiFields.size() == psiFieldsForSetters.size();` in pl/mjedynak/idea/plugins/builder/psi/BuilderPsiClassBuilder.java:79, and change the label of the option by "use single field if possible". In this case the new option will only apply if all needed setters are available.

mjedynak commented:
I think that we would need to display the 'Use single field' box only if all the fields are setters. The information about it would be needed when creating the builder dialog and, if I see correctly, it's not available at the moment. It seems that it would require quite a lot of changes. If you want to go in that direction, I'm willing to review and merge, otherwise feel free to use your fork.

mkickax commented:
@mjedynak, to my mind hiding/showing the checkbox in the first dialog is not a good solution, because we may want to use of single field feature only for a subset of fields (those with setters), even if all fields are not settable by setters.

Here's a last attempt to convince you. What do you think about keeping the checkbox "use single field" and display in the member chooser only fields that have setters? This is similar to the behaviour for outer builder which excludes fields that are not settable by setters or constructors.

mjedynak commented:
I think it's acceptable, but let's keep it unchecked by default.
I also noticed you're using commons-lang lib, which is not available in Community edition.

mkickax commented:
@mjedynak I added the changes we discussed about. I added an extra one also. If using single field + but method, the displayed fields are those with setter AND getter, as we need getter in but method.

I forgot to mention in the commit that I changed the import of commons.lang3 by commons.lang. The latest is already used in the project and should be harmless.

Hope everything is OK. I'm waiting for your code review/comments.
